### PR TITLE
samples: hw_unique_key: Refactor usage of invalid key ID

### PR DESCRIPTION
**Pull request by joerchan**
Refactor sample to use the PSA definitions for invalid key ID. Use status code for checking function success instead of relying on invalid key IDs.

### CODE REVIEW

**Review by mswarowsky — APPROVED** [review]
lgtm, just one nit

while at it could you maybe change the return code of the sample while at it to align it with the other samples:

https://github.com/nrfconnect/sdk-nrf/blob/main/samples/crypto/rsa/src/main.c#L253-L255

File: samples/keys/hw_unique_key/src/derive_key_tfm.c [hunk 1]
```diff
@@ -15,25 +15,27 @@
...
-			uint32_t key_label_len)
+psa_status_t derive_key(psa_key_attributes_t *attributes, uint8_t *key_label,
+			uint32_t key_label_len, psa_key_id_t *key_id_out)
 {
 	psa_status_t status;
 	psa_key_derivation_operation_t op = PSA_KEY_DERIVATION_OPERATION_INIT;
-	psa_key_id_t key_id_out = 0;
+	psa_key_id_t key_id;
+
+	key_id_out = PSA_KEY_ID_NULL;
```
Comment (vili-nordic) [inline samples/keys/hw_unique_key/src/derive_key_tfm.c:25]:
```suggestion
	*key_id_out = PSA_KEY_ID_NULL;
```

Comment (joerchan) [reply samples/keys/hw_unique_key/src/derive_key_tfm.c:25]: Ups, thanks.

**Comment by joerchan** [comment]
> lgtm, just one nit
> 
> while at it could you maybe change the return code of the sample while at it to align it with the other samples:
> 
> https://github.com/nrfconnect/sdk-nrf/blob/main/samples/crypto/rsa/src/main.c#L253-L255

A rather long nit, but done

**Review by mswarowsky — APPROVED** [review]

**Review by vili-nordic — APPROVED** [review]

**Pull request merged**